### PR TITLE
[feat] fetch and display LCA listings

### PR DESCRIPTION
**Pull request by rahihazra**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### :palm_tree: What's new in this PR
[//]: # "REQUIRED - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
This PR fetches all LCA Listings, adjusts the schema for LCA, and displays all LCA Listings on a page (unstyled).

### :evergreen_tree: Screenshots
[//]: # "REQUIRED for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! If you are making changes here, please CC: @kyrenetam at the bottom."

Screen Shot 2024-03-10 at 1 34 35 PM

## :deciduous_tree: How to review
[//]: # 'REQUIRED - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
- order in which to review files:
1. schema.ts
2. limited-case-assistance.ts
3. page.tsx
- should be reviewed all at once (there is only one commit)

## :seedling: Next steps 
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
- styling for displaying the LCA listings

### :information_source: Online sources
[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

### :potted_plant: Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

[//]: # 'This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
CC: @varortz

### CODE REVIEW

File: src/app/onboarding/basic-information/page.tsx [hunk 1]
```diff
@@ -92,4 +92,4 @@ export default function Page() {
       />
     </>
   );
-}
+  */
```
Comment (jinkang-0) [inline src/app/onboarding/basic-information/page.tsx]: can we revert this change? (i.e. add `}` and remove `*/`)
just to make the page work

File: src/types/schema.ts [hunk 2]
```diff
@@ -72,14 +72,15 @@ export interface Interest {
...
   id: UUID;
+  title: Text;
   summary: string;
-  languages: string[];
   country: string;
-  experience_level: string;
   deadline: string;
-  interest_ids: UUID[];
+  research_topic: Text;
+  language: string;
```
Comment (jinkang-0) [inline src/types/schema.ts]: instead of `Text`, let's use `string`
`Text` is more of a `React` markup thing

File: src/types/schema.ts [hunk 3]
```diff
@@ -72,14 +72,15 @@ export interface Interest {
...
+  listing_type: 'Listing Case Assignment';
```
Comment (jinkang-0) [inline src/types/schema.ts]: ~~should we say `"Limited Case Assignment` instead?~~
edit: let's go with what's most up-to-date on the [Typescript schema](https://www.notion.so/calblueprint/Typescript-Schema-07297d182f49405586f5c8232e458051)

File: src/api/supabase/queries/limited-case-assignments.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import supabase from '../createClient';
+
+export async function getAllLCAListings() {
```
Comment (varortz) [inline src/api/supabase/queries/limited-case-assignments.ts]: can we add documentation here? For examples, you can look at the `/queries/interest.ts` file or `/queries/LanguageSupport.ts` file

File: src/types/schema.ts [hunk 5]
```diff
@@ -72,14 +72,15 @@ export interface Interest {
 }
 
 // irrelevant until mvp
```
Comment (varortz) [inline src/types/schema.ts]: hehe let's remove this comment

File: src/app/onboarding/basic-information/styles.tsx [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import styled from 'styled-components';
```
Comment (varortz) [inline src/app/onboarding/basic-information/styles.tsx]: It seems like this file is from a December commit, let's go ahead and remove this file along with the changes requested by jinkang on the `/basic-information/page.tsx` file. 

**Quick reminder:** we wanna make sure to go to `git checkout main` -> `git pull` -> `git checkout -b [new-branch-name]` when creating a new branch, this way our branch is up-to-date with what is on main. It seems like there might have been some merging done, and following the steps above will prevent you from having to do any merging to update your branch.

File: src/types/schema.ts [hunk 7]
```diff
@@ -72,14 +72,15 @@ export interface Interest {
...
-export interface LimitedAssistance {
+export interface LimitedCaseAssignment {
```
Comment (varortz) [inline src/types/schema.ts:75]: could we update the LCA schema to match this schema in this screenshot:
image

File: src/types/schema.ts
Comment (jinkang-0) [inline src/types/schema.ts:1]: let's remove this file and use `schema.d.ts` from main branch

**Review by jinkang-0 — APPROVED** [review]

**Review by varortz — APPROVED** [review]
